src/main/java/software/amazon/cloudformation/resource/Serializer.java

```diff
@@ -76,10 +87,37 @@
         return OBJECT_MAPPER.writeValueAsString(modelObject);
     }
 
+    public <T> String compress(final String modelInput) throws IOException {
+        final Map<String, String> map = new HashMap<>();
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        GZIPOutputStream gzip = new GZIPOutputStream(byteArrayOutputStream);
+        gzip.write(modelInput.getBytes(StandardCharsets.UTF_8));
+        gzip.close();
+        map.put(COMPRESSED, Base64.encodeBase64String(byteArrayOutputStream.toByteArray()));
```

Comment:
How about put additional metadata as how it is compressed?  like `"__COMPRESSED_METHOD__": "gzip_base64"`